File: Renderer/src/main/java/org/turbomedia/turboedit/renderer/websocket/WebsocketAuth.java
package org.turbomedia.turboedit.renderer.websocket;

import com.google.protobuf.InvalidProtocolBufferException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.turbomedia.turboedit.renderer.Envs;
import org.turbomedia.turboedit.shared.protocol.ProtocolAuth;

public class WebsocketAuth {
    private static final Logger logger = LoggerFactory.getLogger(WebsocketAuth.class);

    public static boolean CheckAuthPacket(byte[] data) {
        try {
            var packet = ProtocolAuth.AuthRequest.parseFrom(data);

            return packet.getPassword().strip().equals(Envs.ACCESS_KEY.strip());
        } catch (InvalidProtocolBufferException e) {
            logger.error("{} = {}", e.getClass().getName(), e.getMessage());
            return false;
        }
    }
}

File: Renderer/src/main/kotlin/turboedit/renderer/websocket/WebsocketAuth.kt
package turboedit.renderer.websocket

import com.google.protobuf.InvalidProtocolBufferException
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.turbomedia.turboedit.shared.protocol.ProtocolAuth.AuthRequest
import turboedit.renderer.Envs

object WebsocketAuth {
    private val logger: Logger = LoggerFactory.getLogger(WebsocketAuth::class.java)

    fun checkAuthPacket(data: ByteArray?): Boolean {
        try {
            val packet = AuthRequest.parseFrom(data)

            return packet.getPassword().trim() == Envs.ACCESS_KEY!!.trim()
        } catch (e: InvalidProtocolBufferException) {
            logger.error("{} = {}", e.javaClass.getName(), e.message)
            return false
        }
    }
}
